### PR TITLE
fix: change event fetching algorithm

### PR DESCRIPTION
**Pull request by bilbeyt**

### CODE REVIEW

File: beamer/tests/agent/test_request.py [hunk 1]
```diff
@@ -363,20 +363,6 @@ def test_agent_only_claim_once_after_restart(request_manager, token, agent, dire
                     break
 
 
-def test_event_fetcher_progress_without_events(agent, direction):
```
Comment (fredo) [inline beamer/tests/agent/test_request.py:366]: this test should still pass IMO

File: beamer/tests/agent/test_request.py [hunk 2]
```diff
@@ -363,20 +363,6 @@ def test_agent_only_claim_once_after_restart(request_manager, token, agent, dire
...
-    with Sleeper(5) as sleeper:
-        while ape.chain.chain_id not in agent.get_context(direction).latest_blocks:
-            sleeper.sleep(0.1)
-    for _ in range(5):
-        ape.chain.mine(1)
-        with Sleeper(5) as sleeper:
-            while (
-                agent.get_context(direction).latest_blocks[ape.chain.chain_id].number
-                != ape.chain.blocks[-1].number
```
Comment (fredo) [inline beamer/tests/agent/test_request.py:375]: What does `[-1]` mean in `ape.chain.blocks[-1].number`

File: beamer/tests/agent/test_request.py (hunk 1 above)
Comment (bilbeyt) [reply beamer/tests/agent/test_request.py:366]: actually this test is normally looking for update block number without any new events, and now block number shouldnt increase normally

File: beamer/tests/agent/test_request.py (hunk 2 above)
Comment (bilbeyt) [reply beamer/tests/agent/test_request.py:375]: latest block number in this case

Comment (bilbeyt) [reply beamer/tests/agent/test_request.py:375]: i mean this is `ape.chain.blocks[-1]` latest block

Comment (fredo) [reply beamer/tests/agent/test_request.py:375]: Ah ok so -1 does not mean `latest - 1`?

File: beamer/tests/agent/test_request.py (hunk 1 above)
Comment (fredo) [reply beamer/tests/agent/test_request.py:366]: it should, Even though there aren't new events, new blocks should be updated in fetch()

File: beamer/tests/agent/test_request.py (hunk 2 above)
Comment (bilbeyt) [reply beamer/tests/agent/test_request.py:375]: no directly the latest like in python list

Comment (fredo) [reply beamer/tests/agent/test_request.py:375]: got it, just wanted to be crystal clear. thanks

File: beamer/events.py [hunk 3]
```diff
@@ -355,7 +355,7 @@ def fetch(self) -> list[Event]:
             events = self._fetch_range(from_block, to_block)
             if events is not None:
                 result.extend(events)
-            from_block = BlockNumber(to_block + 1)
+                from_block = BlockNumber(to_block + 1)
```
Comment (istankovic) [inline beamer/events.py:358]: Can we have a test for this, please?

**Review by istankovic — APPROVED** [review]

**Pull request merged**